### ISSUE TITLE
WorkloadIdentityCredentialBuilder / unnecessary param validation

### ISSUE THREAD

**Issue by janchristian-haddorp** [Azure/azure-sdk-for-java#35542]
The build method on [WorkloadIdentityCredentialBuilder](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/src/main/java/com/azure/identity/WorkloadIdentityCredential.java) checks if all params are non null.

```
public WorkloadIdentityCredential build() {
    ValidationUtil.validate(this.getClass().getSimpleName(), LOGGER, "Client ID", clientId,
        "Tenant ID", tenantId, "Service Token File Path", tokenFilePath);

    return new WorkloadIdentityCredential(tenantId, clientId, tokenFilePath, identityClientOptions.clone());
}
```

This is unnecessary and not really user friendly as the class [WorkloadIdentityCredential](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/src/main/java/com/azure/identity/WorkloadIdentityCredential.java) has a perfect fallback handling implemented. If params are null values from env vars will be taken.

```
WorkloadIdentityCredential(String tenantId, String clientId, String federatedTokenFilePath, IdentityClientOptions identityClientOptions) {
    ValidationUtil.validateTenantIdCharacterRange(tenantId, LOGGER);

    Configuration configuration = identityClientOptions.getConfiguration() == null
        ? Configuration.getGlobalConfiguration().clone() : identityClientOptions.getConfiguration();

    String tenantIdInput = CoreUtils.isNullOrEmpty(tenantId)
        ? configuration.get(Configuration.PROPERTY_AZURE_TENANT_ID) : tenantId;

    String federatedTokenFilePathInput = CoreUtils.isNullOrEmpty(federatedTokenFilePath)
        ? configuration.get(AZURE_FEDERATED_TOKEN_FILE) : federatedTokenFilePath;

    String clientIdInput = CoreUtils.isNullOrEmpty(clientId)
        ? configuration.get(Configuration.PROPERTY_AZURE_CLIENT_ID) : clientId;
```

Please remove param validation from builder class.

**Comment by joshfree** [comment]
Thanks for the feedback, @janchristian-haddorp.  /cc @g2vinay @billwert

**Comment by g2vinay** [comment]
@janchristian-haddorp 

Thank you for the feedback.
We can incorporate this for our upcoming release this month.

**Comment by g2vinay** [comment]
@janchristian-haddorp 

To provide an update, The workload Identity credential will be picking up values from environment variables as default.
in out upcoming August release in the coming weeks.

**Comment by g2vinay** [comment]
This feedback is addressed in azure-identity:1.10.0
